### ISSUE TITLE
No compatiple item found with OS version check

### ISSUE THREAD

**Issue by i-n-g-o** [vslavik/winsparkle#284]
Hello.

In Version 0.8.2 with an appcast.xml without XML-tag `<sparkle:minimumSystemVersion>` the OS version gets checked regardless of it not being present in the xml. This seems to be because `item.MinOSVersion = "10";` in function `bool is_compatible_with_windows_version(const Appcast &item_)` found in `appcast.cpp`.

I am not sure if this is wanted behaviour as it would overwrite the minimum OS Version set by `<sparkle:minimumSystemVersion>` from an appcast.

The defines `VER_MAJORVERSION.VER_MINORVERSION.VER_SERVICEPACKMAJOR` are reported to be: `2.1.32` on my system - a check would fail and no appcast item would validate as compatiple with that windows version.

**Comment by i-n-g-o** [comment]
But then `[System.Environment]::OSVersion.Version` in a Powershell reveals:

```
Major  Minor  Build  Revision
-----  -----  -----  --------
10     0      22631  0
```

**Comment by vslavik** [comment]
Fuck.

**Comment by vslavik** [comment]
Fixed 0.8.3 release imminent.

**Comment by i-n-g-o** [comment]
Cool. Thanks for the quick fix!

**Comment by i-n-g-o** [comment]
I am wondering about the use of these defines in that function. Wouldn't these not hardcode the version from the build machine into the library?

I tried this for getting the runtime version - from: https://stackoverflow.com/questions/36543301/detecting-windows-10-version

```
typedef LONG NTSTATUS, *PNTSTATUS;
#define STATUS_SUCCESS (0x00000000)

typedef NTSTATUS (WINAPI* RtlGetVersionPtr)(PRTL_OSVERSIONINFOW);

RTL_OSVERSIONINFOW GetRealOSVersion() {
    HMODULE hMod = ::GetModuleHandleW(L"ntdll.dll");
    if (hMod) {
        RtlGetVersionPtr fxPtr = (RtlGetVersionPtr)::GetProcAddress(hMod, "RtlGetVersion");
        if (fxPtr != nullptr) {
            RTL_OSVERSIONINFOW rovi = { 0 };
            rovi.dwOSVersionInfoSize = sizeof(rovi);
            if ( STATUS_SUCCESS == fxPtr(&rovi) ) {
                return rovi;
            }
        }
    }
    RTL_OSVERSIONINFOW rovi = { 0 };
    return rovi;
}
```

**Comment by vslavik** [comment]
> I am wondering about the use of these defines in that function. Would these not hardcode the version from the build machine into the library?

You misunderstood their meaning (obligatory [RTFM](https://learn.microsoft.com/en-us/windows/win32/api/winnt/nf-winnt-versetconditionmask)). `VerifyVersionInfo` does the same thing as this snipped as discussed in the SO link.

*However*, this drew my attention to incorrect mask and field arguments for `VerifyVersionInfo` — i.e. da4e7e2b5d9f6ca4209ee51fd3190bc06d46f6b1 is incomplete, so thanks for poking...

**Comment by i-n-g-o** [comment]
Ah, I see. Thanks for the link :)

**Comment by vslavik** [comment]
Re-released as 0.8.3.
